lib/puppet/pops/types/type_mismatch_describer.rb

```diff
@@ -581,6 +581,11 @@ def validate_default_parameter(subject, param_name, param_type, value, tense = :
       end
     end
 
+    def get_deferred_function_return_type(value)
+     func =  Puppet::Pops::Loaders.find_loader().load_typed(Puppet::Pops::Loader::TypedName.new(:function, value.name)).value
```

Comment:
This actually won't work as written, I had to specify the module my function was in as a parameter for `find_loader`. I'm sure there are several ways to get the func, some of the funkier than others...